src/utils/carousel.ts

```diff
@@ -0,0 +1,113 @@
+import { Event } from "api/event"
+/**
+ * enum to differentiate the different types of widget we could have
+ * different widget types have different background images
+ */
+export enum WidgetType {
+    LECTURES = 1,
+    EXAMS = 2,
+    NEWS = 3,
+    DEADLINE = 4,
+    CUSTOM = 5,
+}
+
+export interface CarouselItem {
+    /**
+     * number to identify the object and the relative widget
+     */
+    id: number
+    /**
+     * enum field to identify the type of the widget
+     */
+    type: WidgetType
+    /**
+     * string to identify the date of the event(or deadline) contained in the widget
+     */
+    date: string
+    /**
+     * string to identify the time of the event(or deadline) contained in the widget
+     */
+    time: string
+    /**
+     * string to identify the title of the event contained in the widget
+     */
+    title: string
+    /**
+     * string to identify eventually the room when the event takes place
+     */
+    room?: string
+}
+
+/**
+ * Function that helps to format the title of an Highlight. I decided to transform the entire title to lowercase
+ * except for the first letter of each word with more than 3 characters.
+ * @param title
+ * @returns the transformed title
+ */
+export function formatTitle(title: string) {
+    return title
+        .split(" ")
+        .map(item =>
+            item.length > 3
+                ? item.charAt(0).toUpperCase() + item.slice(1).toLowerCase()
+                : item.toLowerCase()
+        )
+        .join(" ")
+}
+
+/**
+ * Function that allows to check the event_type of an event. So far, we filter the events that are lectures, exams or deadlines.
+ * @param typeId of the event
+ * @returns true/false
+ */
+export function checkEventType(typeId: number) {
+    return (
+        typeId === WidgetType.LECTURES ||
+        typeId === WidgetType.EXAMS ||
+        typeId === WidgetType.DEADLINE
+    )
+}
+
+export function createWidget(event: Event) {
+    const days = [
+        "Lunedi",
+        "Martedi",
+        "Mercoledi",
+        "Giovedi",
+        "Venerdi",
+        "Sabato",
+        "Domenica",
+    ]
+    const months = [
+        "Gennaio",
+        "Febbraio",
+        "Marzo",
+        "Aprile",
+        "Maggio",
+        "Giugno",
+        "Luglio",
+        "Agosto",
+        "Settembre",
+        "Ottobre",
+        "Novembre",
+        "Dicembre",
+    ]
+    const dateObj = new Date(event.date_start)
+    const resultDate =
+        days[(dateObj.getDay() - 1 + 7) % 7] +
+        " " +
```

Comment:
Altra cosa non fondamentale, per maggiore chiarezza di calcolo cambierei l'array per conformarsi alla funzione getDay che restituisce un numero da 0 a 6 partendo da Domenica.

```suggestion
    const days = [
        "Domenica",
        "Lunedi",
        "Martedi",
        "Mercoledi",
        "Giovedi",
        "Venerdi",
        "Sabato",
    ]
    const months = [
        "Gennaio",
        "Febbraio",
        "Marzo",
        "Aprile",
        "Maggio",
        "Giugno",
        "Luglio",
        "Agosto",
        "Settembre",
        "Ottobre",
        "Novembre",
        "Dicembre",
    ]
    const dateObj = new Date(event.date_start)
    const resultDate =
        days[dateObj.getDay()] +
        " " +
```

Reply:
giusto si mi ricordo che me l'avevi fatta notare in un audio ma poi mi ero scordato ahah, modifico anche questa grazie